### PR TITLE
Docker image improvements

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes:

## Description
* Do not pass marketplace to native-image when preparing for upload as its redundant
* extract the `DockerImage` class into a different module under the `docker` package and wrap all the `DockerhubClient` functionailty within the docker-image to get info on that docker-image
* use the wrapped docker-image object in all validations

### CODE REVIEW

File: demisto_sdk/commands/validate/default_config.toml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 select = [
-    "BA101", "BA105", "BA106", "BA116", "BA118",
-    "PA107", "PA108", "PA111", "PA115", "PA130",
-    "BC100",
-    "GR106",
-    "DO108",
-    "IN108",
-    "IN130",
-    "PB118",
+#    "BA101", "BA105", "BA106", "BA116", "BA118",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/default_config.toml:4]: Why?

**Review by YuvHayun — APPROVED** [review]
Nice!

File: demisto_sdk/commands/validate/default_config.toml (hunk 1 above)
Comment (GuyAfik) [reply demisto_sdk/commands/validate/default_config.toml:4]: just for testing, thanks for reminding me to remove it, ill revert

Comment (GuyAfik) [reply demisto_sdk/commands/validate/default_config.toml]: reverted :)

**Pull request merged**